src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -774,6 +899,14 @@ def construct_payload(self):
             raise ResourceNotFoundError("The containerapp '{}' does not exist".format(self.get_argument_name()))
 
         self.new_containerapp["properties"] = {}
+
+        if self.get_argument_source() and not self.get_argument_registry_server():
+            registry_server = self.containerapp_def["properties"]["configuration"]["registries"][0]["server"]
```

Comment:
when there is no registry, the code will be broken. Please rerun all tests with --live and I think it will help you find many potential issues

Reply:
This case could occur if the user initially created a Container App with the default `mcr.microsoft.com/k8se/quickstart:latest` image and later tried to update the container app by passing `--source` arg. Example - `az containerapp update -g {rg} -n {name} --source {path}`

Updated this to check for registries first and throw a validation error if it doesn't exist.